include/tvm/expr_operator.h

```diff
@@ -550,6 +551,10 @@ inline Expr MakeConstScalar(Type t, ValueType value) {
   if (t.is_int()) return ir::IntImm::make(t, static_cast<int64_t>(value));
   if (t.is_uint()) return ir::UIntImm::make(t, static_cast<uint64_t>(value));
   if (t.is_float()) return ir::FloatImm::make(t, static_cast<double>(value));
+  if (datatype::Registry::Global()->GetTypeRegistered(static_cast<uint8_t>(t.code()))) {
+    return ir::UIntImm::make(t, datatype::ConvertConstScalar(static_cast<uint8_t>(t.code()),
+                                                             static_cast<double>(value)));
+  }
```

Comment:
We should use FloatImm instead; keep the same type. 
We'll also need to change the lowering pass.

Reply:
Lowering pass will need to lower FloatImm to UIntImm; can use the logic that we have here to convert the FloatImm to the corresponding UIntImm